library/std/src/io/stdio.rs

```diff
@@ -1202,6 +1202,59 @@ where
     }
 }
 
+/// Trait to determine if stdio stream (Stdin, Stdout, Stderr) is a terminal/tty.
+#[unstable(feature = "is_terminal", issue = "80937")]
+pub trait IsTerminal {
```

Comment:
Why are you using a trait ? Are you expecting user code that user code will want to implement that trait ? (if not it must be sealed with a private module).

Reply:
I don't see a scenario in which user code would need to implement this trait, so it could just be converted to regular `impl`s if that is easier. I originally created it as a trait after a recommendation from @bjorn3: https://github.com/rust-lang/rust/pull/81807#discussion_r619015788

Reply:
I do think this should start out as a sealed trait. I can imagine reasons why people *might* want to implement it in the future, but we should be able to make the decision about exposing the trait and its method independently of the decision about letting people implement it outside the standard library.